app/code/BVMyBlog/Blog/Block/LastPosts.php

```diff
@@ -0,0 +1,116 @@
+<?php
+declare(strict_types=1);
+
+namespace BVMyBlog\Blog\Block;
+
+use BVMyBlog\Blog\Model\BlockRepository;
+use Magento\Catalog\Model\Locator\RegistryLocator;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Api\SortOrderBuilder;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\Exception\NotFoundException;
+use Magento\Framework\UrlInterface;
+use Magento\Framework\View\Element\Block\ArgumentInterface;
+use Magento\Framework\View\Element\Template;
+
+/**
+ * Returns 5 latest posts
+ */
+class LastPosts extends Template implements ArgumentInterface
+{
+    /**
+     * @var SortOrderBuilder $sortOrderBuilder
+     */
+    private $sortOrderBuilder;
+
+    /**
+     * @var SearchCriteriaBuilder $searchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * @var BlockRepository $blockRepository
+     */
+    private $blockRepository;
+
+    /**
+     * ScopeConfigInterface $scopeInterface
+     */
+    private $scopeInterface;
+
+    /**
+     * UrlInterface $urlInterface
+     */
+    private $urlBuilder;
+
+    /**
+     * @var RegistryLocator $registryLocator
+     */
+    private $registryLocator;
+
+    /**
+     * @param ScopeConfigInterface $scopeInterface
+     * @param UrlInterface $urlBuilder
+     * @param RegistryLocator $registryLocator
+     * @param SortOrderBuilder $sortOrderBuilder
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param BlockRepository $blockRepository
+     */
+    public function __construct(
```

Comment:
You are extending from Template block, please call its constructor and define constructor arguments